src/main/scala/firrtl/passes/InferTypes.scala

```diff
@@ -97,6 +98,25 @@ object InferTypes extends Pass {
   }
 }
 
+/** Internal node used by the parser for the `[...]` and `[... : ...]` syntax.
+  *  CHIRRTL level type inference then converts this into either a `bits(..., ... , ...)`,
+  *  or [[ir.SubIndex]] depending on the inferred type of the inner expression.
+  */
+case class WSliceNode(expr: ir.Expression, hi: Int, lo: Int) extends ir.Expression {
```

Comment:
Would it be better to define this as a node in `IR.scala` so that it can be more easily accessed from the Chisel converter?

Reply:
No, this is an internal node used to hack around a short coming in the parser (that it does not do type inference). From Chisel you should know whether you want a bit-extract or a vector access since the types are available in the Chisel data structure.